optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/shift/ShiftRestServiceImpl.java

```diff
@@ -115,18 +130,60 @@ public Boolean removeShift(Integer tenantId, Long id) {
     }
 
     @Override
-    public List<ShiftTemplate> getShiftTemplateList(Integer tenantId) {
-        TypedQuery<ShiftTemplate> q = entityManager.createNamedQuery("ShiftTemplate.findAll", ShiftTemplate.class);
-        q.setParameter("tenantId", tenantId);
-        return q.getResultList();
+    public RotationView getRotation(Integer tenantId) {
+        List<ShiftTemplate> shiftTemplateList = entityManager.createNamedQuery("ShiftTemplate.findAll", ShiftTemplate.class)
+                .setParameter("tenantId", tenantId)
+                .getResultList();
+        RotationView out = new RotationView();
+        out.setTenantId(tenantId);
+        out.setSpotList(spotRestService.getSpotList(tenantId));
+        out.setEmployeeList(employeeRestService.getEmployeeList(tenantId));
+        out.setRotationLength(entityManager.createNamedQuery("RosterState.find", RosterState.class)
+                .setParameter("tenantId", tenantId)
+                .getSingleResult().getRotationLength());
+        Map<Long, List<ShiftTemplateView>> spotIdToShiftTemplateViewListMap = new HashMap<>();
+        shiftTemplateList.forEach((shiftTemplate) -> {
+            spotIdToShiftTemplateViewListMap.computeIfAbsent(shiftTemplate.getSpot().getId(),
+                    (k) -> new ArrayList<>())
+                    .add(new ShiftTemplateView(out.getRotationLength(), shiftTemplate));
+        });
+        out.setSpotIdToShiftTemplateViewListMap(spotIdToShiftTemplateViewListMap);
+        return out;
     }
 
     @Override
     @Transactional
-    public void updateShiftTemplate(Integer tenantId, List<ShiftTemplate> shiftTemplateList) {
-        List<ShiftTemplate> oldShiftTemplateList = getShiftTemplateList(tenantId);
+    public void updateRotation(Integer tenantId, RotationView rotationView) {
```

Comment:
This update the entire rotation in batch?

Reply:
Yes